### PR TITLE
all: Improve make lint command

### PR DESCRIPTION
**Pull request by cbornet**
* Removed `ruff check --select I` as `I` is already selected and checked in the main `ruff check` command
* Added checks for non-empty `PYTHON_FILES`
* Run `ruff check` only on `PYTHON_FILES`

### CODE REVIEW

**Comment by cbornet** [comment]
It seems one of the CI runner is out of memory.
Can sonething be done about it ?

**Review by efriis — APPROVED** [review]

File: libs/cli/langchain_cli/integration_template/Makefile [hunk 1]
```diff
@@ -30,14 +30,13 @@ lint_tests: PYTHON_FILES=tests
 lint_tests: MYPY_CACHE=.mypy_cache_test
 
 lint lint_diff lint_package lint_tests:
-	poetry run ruff .
-	poetry run ruff format $(PYTHON_FILES) --diff
-	poetry run ruff --select I $(PYTHON_FILES)
-	mkdir -p $(MYPY_CACHE); poetry run mypy $(PYTHON_FILES) --cache-dir $(MYPY_CACHE)
+	[ "$(PYTHON_FILES)" = "" ] || poetry run ruff $(PYTHON_FILES)
```
Comment (baskaryan) [inline libs/cli/langchain_cli/integration_template/Makefile:33]: should this be `run ruff check ...`

**Comment by efriis** [comment]
bypassing ai21 - known issue in their underlying sdk

File: libs/cli/langchain_cli/integration_template/Makefile (hunk 1 above)
Comment (cbornet) [reply libs/cli/langchain_cli/integration_template/Makefile:33]: Yes, it think it should.
I can open another PR for it.

**Pull request merged**